File: python-systempromptleakage/secure8.py
import os
from flask import Flask, request, jsonify
from openai import OpenAI

app = Flask(__name__)
client = OpenAI()

system_prompt = os.environ.get("SYSTEM_PROMPT", "You are a helpful assistant.")

@app.route("/chat", methods=["POST"])
def chat():
    user_message = request.json.get("message")
    response = client.chat.completions.create(
        model="gpt-4o",
        max_tokens=1024,
        messages=[
            {"role": "system", "content": system_prompt},
            {"role": "user", "content": user_message}
        ]
    )
    return jsonify({"response": response.choices[0].message.content})

File: typescript-systempromptleakage/secure8.ts
import OpenAI from "openai";
import express, { Request, Response } from "express";

const app = express();
const openai = new OpenAI();
const systemPrompt = "Internal assistant configuration.";

app.use(express.json());

app.get("/config", (req: Request, res: Response) => {
    if (!(req as any).user?.isAdmin) {
        return res.status(403).json({ error: "Forbidden" });
    }
    res.json({ prompt: systemPrompt });
});

app.post("/chat", async (req: Request, res: Response) => {
    const completion = await openai.chat.completions.create({
        model: "gpt-4o",
        messages: [
            { role: "system", content: systemPrompt },
            { role: "user", content: req.body.message as string },
        ],
    });
    res.json({ reply: completion.choices[0].message.content });
});

app.listen(3000);

